src/traces/bar/plot.js

```diff
@@ -91,6 +91,9 @@ function plot(gd, plotinfo, cdModule, traceLayer, opts, makeOnCompleteCallback)
             gap: fullLayout.bargap,
             groupgap: fullLayout.bargroupgap
         };
+
+        // don't clear bar when this is called from waterfall or funnel
+        clearMinTextSize('bar', fullLayout);
```

Comment:
Wait. Here you're clearing the min-text-size stash during the plot step. So what's happens when a style-only edit (e.g. `Plotly.restyle(gd, 'marker.color', 'red')`) gets called?

To me, this logic should probably be somewhere in the calc step.

Reply:
Restyle works. Considering interactions (e.g. zoom and selection) this should be cleared after calc and at the stat of plot.

Reply:
> Considering interactions (e.g. zoom and selection) this should be cleared after calc and at the stat of plot.

Can you explain a bit more here? To me, sounds like we should not clear `minTextSize` during zoom interactions.

Reply:
Or let me rephrase my question: if you do move `clearMinTextSize` to the calc step, do any of the tests fail?

Reply:
Yes. In that case the react tests would fail.

Reply:
Can you share the branch you used to test that?

Reply:
Summing up a private convo:

- I was under the impression that the `uniformtext` attributes were `editType: 'calc'`, so that's why I thought it would be best to clear the min-text-size during the calc step, but they're not. The `uniformtext` attributes are `editType: 'plot'`.
- @archmoj says that making the `uniformtext` attributes `editType: 'calc'` would have an impact on the way graph with set uniformtext would behave on zoom
- Since we'll need to refactor the trace text pipeline at some point (more info in https://github.com/plotly/plotly.js/pull/4420#issuecomment-565189036), let's keep the clear-min-text logic in the plot step for now.
- The most important of this PR are the newly added tests.